src/super_gradients/training/utils/config_utils.py

```diff
@@ -63,9 +64,23 @@ def get_unused_params(self) -> Set[str]:
         unused_params = self.get_all_params() - self.get_used_params()
         return unused_params
 
+    def __copy__(self):
+        cls = self.__class__
+        result = cls.__new__(cls)
+        result.__dict__.update(self.__dict__)
+        return result
+
+    def __deepcopy__(self, memo):
+        cls = self.__class__
+        result = cls.__new__(cls)
+        memo[id(self)] = result
+        for k, v in self.__dict__.items():
+            setattr(result, k, deepcopy(v, memo))
+        return result
+
 
 class AccessCounterDict(Mapping, AccessCounterMixin):
-    def __init__(self, config: Mapping, access_counter: Mapping[str, int] = None, prefix: str = ""):
+    def __init__(self, config: dict, access_counter: Mapping[str, int] = None, prefix: str = ""):
```

Comment:
why was that changed to `dict`?
we want to support other types as well (like `DictConfig` that does not inherit from dict)

Reply:
`Mapping` doesn't necessarily support item assignment. I add DictConfig in the typing: 4afca0dea77e2ea98c6bd3ca6b51b38e2dc970bc